src/js/components/Menu/Menu.js

```diff
@@ -23,175 +24,249 @@ const ContainerBox = styled(Box)`
 
-  constructor(props) {
-    super(props);
+To navigate within menu:
+- Up/down arrow keys can be used and will loop through options (keeping focus within the Menu)
+- Tab can be used, but once the last menu item is reached, Tab will close the Menu and continue through page content.
 
-    this.state = { activeItemIndex: -1, open: props.open || false };
-  }
+To close the menu:
+- Tabbing beyond the first or last menu item.
+- Esc will close the menu
+- Select a menu item
+
+To make a selection:
+- Enter key is pressed.
+- Space is pressed.
+
+*/
+
+const Menu = forwardRef((props, ref) => {
+  const {
+    children,
+    disabled,
+    dropAlign,
+    dropBackground,
+    dropProps,
+    dropTarget,
+    // forwardRef,
+    justifyContent,
+    icon,
+    items,
+    label,
+    messages,
+    onKeyDown,
+    open,
+    plain,
+    size,
+    theme,
+    ...rest
+  } = props;
+  const MenuIcon = theme.menu.icons.down;
+  const iconColor = normalizeColor('control', theme);
+  const align = dropProps.align || dropAlign;
+  const buttonRefs = {};
+  let timeoutID; // to track if focus is within the menu items, see menuOnFocus/menuOnBlur
+
+  const [activeItemIndex, setActiveItemIndex] = useState(-1);
+  const [isOpen, setOpen] = useState(open || false);
+  const [isManagingFocus, setManagingFocus] = useState(false);
+
+  useEffect(() => {
+    if (isOpen) {
+      setManagingFocus(true);
+      if (isManagingFocus) {
+        if (align.top === 'bottom') {
+          buttonRefs[0].focus();
+          setActiveItemIndex(0);
+        } else {
+          buttonRefs[items.length].focus();
+          setActiveItemIndex(-1);
+        }
+      }
+    }
+  }, [isManagingFocus]);
 
-  onDropClose = () => {
-    this.setState({
-      activeItemIndex: -1,
-      open: false,
-    });
+  const onDropClose = () => {
+    setActiveItemIndex(-1);
+    setOpen(false);
+    setManagingFocus(false);
   };
 
-  onSelectMenuItem = event => {
-    const { activeItemIndex } = this.state;
+  const onDropOpen = () => {
+    setOpen(true);
+    setManagingFocus(true);
+  };
+
+  const onSelectMenuItem = event => {
     if (activeItemIndex >= 0) {
       event.preventDefault();
       event.stopPropagation();
-      this.buttonRefs[activeItemIndex].click();
+      buttonRefs[activeItemIndex].click();
     }
   };
 
-  onNextMenuItem = event => {
+  const onNextMenuItem = event => {
     event.preventDefault();
-    const { activeItemIndex, open } = this.state;
-    if (!open) {
-      this.setState({
-        open: true,
-        activeItemIndex: -1,
-      });
+    if (!isOpen) {
+      onDropOpen();
     } else {
-      const { items } = this.props;
-      const index = Math.min(activeItemIndex + 1, items.length - 1);
-      this.setState({ activeItemIndex: index });
-      // this.setState({ activeSuggestionIndex: index },
-      //   this._announceSuggestion.bind(this, index));
+      let index;
+      switch (align.top) {
+        case 'bottom':
```

Comment:
Not sure why we need `switch` for a single case, is there a particular reason you chose it?

Reply:
I think initially I thought there were going to be more cases, but in the end there weren't. I can simplify it and push the changes.